### DISCUSSION TITLE
HA causing Battery to stop discharging

### DISCUSSION THREAD

**Discussion by willbiker** [wills106/homeassistant-solax-modbus#603] [General]
Hi All.
I have used HA with this integration for almost a year to interact with my Solax inverter and its awesome! However for the past few months, I would notice that my battery would randomly stop discharging even if there is Battery charge available. After rebooting the inverter, normal operation would resume.  In order to troubleshoot, I switched off my Waveshare converter for 3 weeks and I had no issues at all. As soon as I switched it back on, the issue returned.

I am a basic user of HA, its still very new to me but I checked the logs and I can see that the integration log would occasionally show basic connectivity issues (error reading input registers) and then sometimes specific errors related to a single register and then sometimes it would send a whole bunch of settings to the inverter in one large send. See attached screenshots. What do you suggest the issue could be?

Thanks in advance

Will

![Screenshot_20231025_200158_Home Assistant](https://github.com/wills106/homeassistant-solax-modbus/assets/117373918/40f7082d-44ff-4577-9b61-e86efb610732)
![Screenshot_20231025_223940_Home Assistant](https://github.com/wills106/homeassistant-solax-modbus/assets/117373918/f0f6efb1-4a8b-4f04-9005-21d368692a0c)
![Screenshot_20231103_161453_Home Assistant](https://github.com/wills106/homeassistant-solax-modbus/assets/117373918/3016166f-0c21-4e56-8e16-cd5c0f055c3f)
![Screenshot_20231103_161458_Home Assistant](https://github.com/wills106/homeassistant-solax-modbus/assets/117373918/4ce01774-f894-4b66-a27b-add697861bee)
![Screenshot_20231103_161502_Home Assistant](https://github.com/wills106/homeassistant-solax-modbus/assets/117373918/f144a5ce-e95e-4b4d-85bf-e7619595e5b6)
![Screenshot_20231103_161547_Home Assistant](https://github.com/wills106/homeassistant-solax-modbus/assets/117373918/ef76af00-efbc-45a4-a175-7c7757c315b2)

**Comment by willbiker** [comment]
Bump :)

**Comment by cleveHEX** [comment]
No bump will help you, only make anyone less willing to help you, this is not a commercial service. I didn't bother to look at your problem, because you were too lazy to copy the error logs instead of screenshoting them without cropping. Also no configurations provided.

**Reply by willbiker** [reply]
Thanks so much for your kind words and for creating such a sense of warmth in this community. I will try to work it out on my own.

**Reply by cleveHEX** [reply]
I have put more effort than you to my message. You put the least effort possible and expect everyone to solve the problem to you in a few hours.

**Reply by alienatedsec** [reply]
@wills106 I am not sure what is going on with your repo or if you allow this to carry on. @cewbdex its a second time where I find your comments either upsetting or unhelpful. 

Again, please hold your ambitions and help the community. Your offensive comments are not necessary.

**Comment by willbiker** [comment]
I posted on the off chance that it would be obvious to someone but I have a few things I can try so I will give them a go and if I fail, I will come back with a more friendly error log snippet. Thanks

**Comment by wills106** [comment]
@cewbdex can you try and be less abrupt please. It would have been more helpful to explain how to find the correct log info.
@willbiker if you look in the issue tracker https://github.com/wills106/homeassistant-solax-modbus/issues/new/choose it states how to find the correct log information.

People need to understand it takes time for me to catch up on issues and discussion and can take a few days or even longer to reply.

**Reply by cleveHEX** [reply]
I acknowledge your request, I will try or not post at all.

**Comment by wills106** [comment]
I'm a bit confused as to what Inverter you have.
You are getting an error for "backup_gridcharge" which is a Gen3 only sensor.
But the Integration seems to be also returning "selfuse_discharge_min_soc" which is a Gen4 only sensor.

Have you made any changes to the code? Like adding in an unrecognised Inverter's serial number?

**Reply by willbiker** [reply]
Sorry for the late reply, I somehow missed the notification. I have a X1-G4 HYBRID 6.0. I haven't made any code changes, I have set up some routines which are triggered from a button for setting the charge from grid settings however these have been disabled for a number of months so cant be the issue. I wondered if it could be caused by a connectivity issue as I am using a powerline to connect my RS converter to the network, however I rarely see a connection issue when the issue occurs. The battery stops discharging but the connection remains ok to the inverter. It may be simpler for me to start a fresh and reinstall the integration...
